ballerina/channel_csv_io.bal

```diff
@@ -40,34 +40,71 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][] content) returns Error? {
+isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
     WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
+    if (content is string[][]){
     foreach string[] r in content {
         var writeResult = csvChannel.write(r);
         if writeResult is Error {
             check csvChannel.close();
             return writeResult;
+            }
+        }
+    }
+    else if(content is map<anydata>[]) {
+        foreach map<anydata> row in content {
+            string[] sValues = [];
+            foreach [string, anydata] [subject, value] in row.entries() {
```

Comment:
Is the order preserved here? Better to verify

Reply:
it is checked in tests similar to [this](https://github.com/sahanHe/module-ballerina-io/blob/63f4c94fb742b3ee4e6946aeb44fb0b812897602/ballerina/tests/csv_io.bal#L194)

Reply:
Ack